File: VB/ChartLesson3/Models/GspDbContext.vb
Imports System.Data.Entity

Partial Public Class GspDbContext
    Inherits DbContext

    Public Sub New()
        MyBase.New("name=GspDbContext")
    End Sub

    Public Overridable Property GSPs As DbSet(Of GSP)

    Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
        modelBuilder.Entity(Of GSP)() _
            .Property(Function(e) e.SSMA_TimeStamp) _
            .IsFixedLength()
    End Sub
End Class

File: CS/ChartLesson3/Models/GspDbContext.cs
namespace ChartLesson3.Models {
    using System;
    using System.Data.Entity;
    using System.ComponentModel.DataAnnotations.Schema;
    using System.Linq;

    public partial class GspDbContext : DbContext {
        public GspDbContext()
            : base("name=GspDbContext") {
        }

        public virtual DbSet<GSP> GSPs { get; set; }

        protected override void OnModelCreating(DbModelBuilder modelBuilder) {
            modelBuilder.Entity<GSP>()
                .Property(e => e.SSMA_TimeStamp)
                .IsFixedLength();
        }
    }
}
